src/contracts/Protocol/QuestionAPI.sol

```diff
@@ -19,12 +20,14 @@ contract QuestionAPI is Ownable, NFTLocked {
         address bountyQuestion,
         address questionStateController,
         address claimController,
-        address costController
+        address costController,
+        address vault
     ) {
         _question = BountyQuestion(bountyQuestion);
         _questionStateController = IQuestionStateController(questionStateController);
         _claimController = IClaimController(claimController);
         _costController = IActionCostController(costController);
+        _vault = Vault(vault);
```

Comment:
should this be on the `CostController` instead of the API?  not sure which is better

Reply:
I was going back and forth but I think the cost controller might make more sense